packages/backend/src/asset/service.ts

```diff
@@ -99,13 +108,30 @@ async function createAsset(
   }: CreateOptions
 ): Promise<Asset | AssetError> {
   try {
-    // check if exists but deleted | by code-scale
-    const deletedAsset = await Asset.query(deps.knex)
-      .whereNotNull('deletedAt')
-      .where('code', code)
-      .andWhere('scale', scale)
+    const assets = await Asset.query(deps.knex)
       .andWhere('tenantId', tenantId)
-      .first()
+      .select('*')
+
+    const sameCodeAssets = assets.find((asset) => asset.code === code)
+    if (!sameCodeAssets && assets.length > 0) {
+      const exchangeUrlSetting = await deps.tenantSettingService.get({
+        tenantId,
+        key: TenantSettingKeys.EXCHANGE_RATES_URL.name
+      })
+
+      const tenantExchangeRatesUrl = Array.isArray(exchangeUrlSetting)
```

Comment:
I wonder why the type returned is `TenantSetting | TenantSetting[]` I think it should always be `TenantSetting[]`, unless I'm missing something

CC @golobitch

Reply:
hm,...good question. I don't remember why is there TenantSetting as a return type. @sanducb I think that you can change the return type of that to always return array.